--- main.org.c	2015-05-25 22:38:11.000000000 +0200
+++ main.c	2015-05-25 22:38:36.000000000 +0200
@@ -4187,16 +4187,19 @@ static inline void SchedulerBasicP__Sche
     {
       uint8_t nextTask;
 
-      { __nesc_atomic_t __nesc_atomic = __nesc_atomic_start();
-        {
-          while ((nextTask = SchedulerBasicP__popTask()) == SchedulerBasicP__NO_TASK) 
-            {
-              SchedulerBasicP__McuSleep__sleep();
-            }
-        }
-#line 161
-        __nesc_atomic_end(__nesc_atomic); }
-      SchedulerBasicP__TaskBasic__runTask(nextTask);
+//       { __nesc_atomic_t __nesc_atomic = __nesc_atomic_start();
+//         {
+//           while ((nextTask = SchedulerBasicP__popTask()) == SchedulerBasicP__NO_TASK)
+//             {
+//               SchedulerBasicP__McuSleep__sleep();
+//             }
+//         }
+// #line 161
+//         __nesc_atomic_end(__nesc_atomic); }
+      if((nextTask = SchedulerBasicP__popTask()) != SchedulerBasicP__NO_TASK) {
+        SchedulerBasicP__TaskBasic__runTask(nextTask);
+      }
+      cycles++;
     }
 }
 
